src/mp4box/trun.rs

```diff
@@ -83,6 +84,16 @@ impl<R: Read + Seek> ReadBox<&mut R> for TrunBox {
 
         let (version, flags) = read_box_header_ext(reader)?;
 
+        let other_size = size_of::<u32>() // sample_count
+            + if TrunBox::FLAG_DATA_OFFSET & flags > 0 { size_of::<i32>() } else { 0 } // data_offset
+            + if TrunBox::FLAG_FIRST_SAMPLE_FLAGS & flags > 0 { size_of::<u32>() } else { 0 }; // first_sample_flags
+        let sample_size = if TrunBox::FLAG_SAMPLE_DURATION & flags > 0 { size_of::<u32>() } else { 0 } // sample_duration
+            + if TrunBox::FLAG_SAMPLE_SIZE & flags > 0 { size_of::<u32>() } else { 0 } // sample_size
+            + if TrunBox::FLAG_SAMPLE_FLAGS & flags > 0 { size_of::<u32>() } else { 0 } // sample_flags
+            + if TrunBox::FLAG_SAMPLE_CTS & flags > 0 {
+                if version == 0 { size_of::<u32>() } else { size_of::<i32>() } // sample_composition_time_offset
```

Comment:
Can probably just collapse this into `size_of::<u32>()`.

Reply:
fixed in [619a36b](https://github.com/privacyresearchgroup/mp4-rust/pull/2/commits/619a36b38b435b09ba997434d810f0eab2ad3b35)